ext/mysql2/client.c

```diff
@@ -1425,7 +1429,9 @@ void init_mysql2_client() {
   rb_define_private_method(cMysql2Client, "write_timeout=", set_write_timeout, 1);
   rb_define_private_method(cMysql2Client, "local_infile=", set_local_infile, 1);
   rb_define_private_method(cMysql2Client, "charset_name=", set_charset_name, 1);
+#if defined(MYSQL_SECURE_AUTH)
   rb_define_private_method(cMysql2Client, "secure_auth=", set_secure_auth, 1);
+#endif
```

Comment:
For consistency with several other cases, let's allow this to fail silently by leaving the method in place, but turning it into a no-op.

Reply:
Removed ifdef here.  `set_secure_auth` method already handles if `MYSQL_SECURE_AUTH` is defined or not, here we do not need to handle it again. please give additional comment If I do not understand your review comment correctly.